accord-core/src/main/java/accord/local/cfk/Utils.java

```diff
@@ -261,6 +262,28 @@ static TxnId[] removePrunedAdditions(TxnId[] additions, int additionCount, TxnId
         return prunedIds;
     }
 
+    static TxnId[] removeUnmanaged(TxnId[] ids)
+    {
+        int count = 0;
```

Comment:
Nit: if we have mostly-managed, we could just pre-allocate an array with same size as ids, and copy as we go, then system.arrays copy the `count` elements. Feels it might be a bit more efficient than iterating twice.

Reply:
I will make a note to consider this more later. My intuition is that counting will have negligible cost, as these arrays should normally be small. But equally in C* we expect to have very few unmanaged transactions.

Reply:
Hmm, actually no I think the current approach is optimal in the case we expect it to be mostly managed, as we don't copy anything if we don't remove anything?